### DISCUSSION TITLE
Encoding of the JSON Sends API request

### DISCUSSION THREAD

**Discussion by DonSYS91** [dani-garcia/vaultwarden#3029] [General]
Hello, 
I'm trying to use the Sends function programmatically and so far I got the API authentication to work but still failing on understanding how the API request is encoded.
Looking at the following JSON when submitting a Send via the browser:

```{
	"type": 0,
	"name": "2.4GuePSqfjpFy1Ju6cH0QmQ==|UtBGRx5bxZtHwluw35YUGg==|o+3O41qtF/P8mGg3sY9N5yxAMU8BiLhw03XYYEQMhf4=",
	"notes": null,
	"maxAccessCount": null,
	"expirationDate": null,
	"deletionDate": "2022-12-28T22:07:20.000Z",
	"key": "",
	"disabled": false,
	"hideEmail": true,
	"text": {
		"response": null,
		"text": "2.FoM6wNYe8EWKKzG7kltsyg==|n9Nkbs795pfE7CYf81q0tA==|Yd3YQTkpMD7AoiChaNsfYhq2IJ6+B8YVtpUOIU9+SQ4=",
		"hidden": true
	}
}```

I need to construct the name and text filled on my own but tried to decode base64 but failed so far, I also removed the key part for the mentioned example here

Can anyone help me out please?

Thanks

**Comment by BlackDex** [comment]
Just to be clear, Vaultwarden it self doesn't do anything regarding encryption it self in regards to the vault data.

I suggest to read https://bitwarden.com/help/what-encryption-is-used/ regarding the encryption and description.

**Comment by jjlin** [comment]
Instead of using the private API directly, you should probably be using the CLI API:

https://bitwarden.com/blog/bringing-restful-api-to-the-bitwarden-cli/

**Comment by DonSYS91** [comment]
Thank you so much @BlackDex and @jjlin for the blazing fast response.
the request was accepted by the server when I tried to encrypt aes-cbc 256 with the same key I had from the request, but looks my encryption wasn't correct since the server couldn't decode the name.

Anyway I wasn't aware at all the bw cli can expose an API, I will use that now.

Thanks again
